javalib/src/main/scala/java/nio/file/Files.scala

```diff
@@ -92,17 +92,16 @@ object Files {
       else classOf[PosixFileAttributes]
 
     val attrs = Files.readAttributes(source, attrsCls, linkOpts)
-    if (attrs.isSymbolicLink())
-      throw new IOException(
-        s"Unsupported operation: copy symbolic link $source to $target"
-      )
-
     val targetExists = exists(target, linkOpts)
     if (targetExists && !options.contains(REPLACE_EXISTING))
       throw new FileAlreadyExistsException(target.toString)
 
     if (isDirectory(source, Array.empty)) {
       createDirectory(target, Array.empty)
+    } else if (attrs.isSymbolicLink() &&
+        options.contains(LinkOption.NOFOLLOW_LINKS)) {
+      if (targetExists) Files.delete(target)
```

Comment:
Sorry to worry & run.

The unconditional `delete()` here seems suspicious.  I know that
it is in the code being replaced.  It is also probably a "OK on unix but needed on Windows kind of thing".

Seems like the delete should be guarded by the option REPLACE_EXISTING as is done several lines below.

Reply:
We can enter the `if (targetExists) Files.delete(target)` only if `REPLACE_EXISTING` is passed and it actually exists, so we don't really need to check it one more time. Otherwise it was already validated - if it exists and there is not `REPLACE_EXISTING` then we would throw exception before reaching this line

Reply:
Thank you for explaining.

Now that you point it out, I now see the code on lines 96 & 97.
```
    if (targetExists && !options.contains(REPLACE_EXISTING))
      throw new FileAlreadyExistsException(target.toString)
```
I also traced through why the ` if (targetExists) Files.delete(target)` is necessary.

I need to ask Ariadne for more string or I will never make it out of this maze.